repo-agent/issue-sandbox/main.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+)
+
+func main() {
+	cmdCodeSrv, err := startCodeServer()
+	if err != nil {
+		log.Fatalf("failed to start code-server: %v", err)
+	}
+	defer func() {
+		if cmdCodeSrv.Process != nil {
+			_ = cmdCodeSrv.Process.Kill()
+		}
+	}()
+
+	err = runIssueSolver()
+	if err != nil {
+		log.Fatalf("failed solving issue: %v", err)
+	}
+
+	err = cmdCodeSrv.Wait()
+	if err != nil {
+		log.Printf("Code Server exited with error: %v", err)
+	} else {
+		log.Println("Code Server exited with no error")
+	}
+}
+
+func runIssueSolver() error {
+	log.Println("Starting issue solver")
+
+	// Environment variables
+	gitPushEnabled := os.Getenv("GIT_PUSH_ENABLED") == "true"
```

Comment:
Using `log.Fatalf` for error handling can be abrupt as it calls `os.Exit(1)`. Consider returning errors and handling them in the `main` function. This allows for more graceful shutdowns and makes the functions more reusable and testable.